dashboard/streamlit.py

```diff
@@ -0,0 +1,330 @@
+    live_metrics = get_latest_metrics(cursor)
+    archival_metrics = get_archival_data(cursor)
+
+    # Add filter
+    filter_plant = get_plant_filter(list(live_metrics['plant_name']))
+    st.write(" ")
+
+    # Filter data
+    filtered_data = filter_by_plant(
+        filter_plant, live_metrics, archival_metrics)
+
+    # Display charts and tables
+    display_charts(filtered_data[0], filtered_data[1])
+
+
+def display_charts(data_live: pd.DataFrame, data_archival: pd.DataFrame) -> None:
+    """Function to display and format charts."""
+    temp, name_id = st.columns((6, 1))
+    with temp:
+        st.altair_chart(overlay_temperature_chart(data_live,
+                                                  data_archival))
+    with name_id:
+        table_data = get_data_plant_table(data_live)
+        st.write(
+            table_data.style.set_table_styles([
+                {'selector': 'th', 'props': [('text-align', 'left')]},
+                {'selector': 'td', 'props': [('width', '100px')]},
+                {'selector': 'col1', 'props': [('width', '200px')]}
+            ])
+        )
+    st.write(" ")
+
+    st.altair_chart(overlay_soil_moisture_chart(data_live,
+                                                data_archival))
+
+    st.altair_chart(plot_last_watered(data_live))
+
+
+def get_plant_filter(plant_names: list, key: str = "plant_filter") -> list:
+    """Create a multiselect filter based on plant names."""
+    return st.multiselect(
+        'Filter by Plant(s)',
+        options=plant_names,
+        default=plant_names,
+        key=key
+    )
+
+
+def filter_by_plant(selected_plants: list, plant_metrics: pd.DataFrame, archive_plants: pd.DataFrame) -> list:
+    """Filter visualizations based on selected plant names."""
+    if selected_plants:
+        return [plant_metrics[plant_metrics['plant_name'].isin(selected_plants)],
+                archive_plants[archive_plants['plant_name'].isin(selected_plants)]]
+    return [plant_metrics, archive_plants]
+
+
+def get_latest_metrics(cursor: Cursor) -> pd.DataFrame:
+    """Function gets the latest plant health metrics including: temperature, soil moisture levels
+    plant name, time of recording and last_watered, and extracts these to a dataframe."""
+    query = f"""
+        SELECT pm.temperature, pm.soil_moisture, latest_recording_info.latest_time,
+          p.plant_name, pm.plant_id, pm.last_watered
+        FROM {environ['SCHEMA_NAME']}.plant_metric pm
+        JOIN (SELECT plant_id,
+            MAX(recording_taken) as latest_time
+            FROM {environ['SCHEMA_NAME']}.plant_metric
+            GROUP BY plant_id) as latest_recording_info
+        ON pm.plant_id = latest_recording_info.plant_id
+            AND pm.recording_taken = latest_recording_info.latest_time
+        JOIN {environ['SCHEMA_NAME']}.plant as p ON pm.plant_id = p.plant_id;
+        """
+    cursor.execute(query)
+    result = cursor.fetchall()
+
+    return pd.DataFrame(result)
```

Comment:
It would be great if we refactor and put all queries and DB connections into a separate directory!